#!/usr/bin/env python

PACKAGE = 'mixmcl'

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t, double_t, str_t, bool_t

gen = ParameterGenerator()

########################Dual MCL########################
gen.add("feature_resolution_x" , int_t, 0, "MixMCL parameters", 10, 1, 100) 
gen.add("feature_resolution_y" , int_t, 0, "MixMCL parameters", 10, 1, 100) 
gen.add("feature_resolution_d" , int_t, 0, "MixMCL parameters", 10, 1, 100) 
gen.add("sample_param_filename", str_t, 0, "MixMCL parameters") 
gen.add("dual_normalizer_ita"  , double_t, 0, "MixMCL parameters", 0.0000001, 0.0, 0.1) 
gen.add("mixing_rate"          , double_t, 0, "MixMCL parameters", 0.1, 0.0, 1.0) 
gen.add("dual_loc_bandwidth"   , double_t, 0, "MixMCL parameters", 10.0, 0.0, 100.0) 
gen.add("dual_ori_bandwidth", double_t, 0, "MixMCL parameters", 0.4, 0.0, 10.0) 
######################end Dual MCL######################

exit(gen.generate(PACKAGE, "mixmcl_node", "MIXMCL"))
